src/main/java/org/testng/TestNG.java

```diff
@@ -947,9 +947,11 @@ private void addReporter(Class<? extends IReporter> r) {
   }
 
   private void initializeDefaultListeners() {
-    m_testListeners.put(ExitCodeListener.class, new ExitCodeListener(this));
+    ExitCodeListener exitCodeListener = new ExitCodeListener(this);
+    m_testListeners.put(ExitCodeListener.class, exitCodeListener);
 
     if (m_useDefaultListeners) {
+      addListener((ITestNGListener) exitCodeListener);
```

Comment:
Why do you add it only here?

Reply:
I didn't quite get you there. Can you please clarify ?